spec/models/customer_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+RSpec.describe Customer, type: :model do
+  subject { Customer.new(first_name: "Jack", last_name: "Smith", phone: "8889995678", email: "jsmith@sample.com" )}
+  it "is valid with valid attributes" do
+    expect(subject).to be_valid
+  end
+  it "is not valid without a first_name" do
+    subject.first_name=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without a last_name" do
+    subject.last_name=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without a phone number" do
+    subject.phone=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without an email" do
+    subject.email=nil
+    expect(subject).to_not be_valid
+end
+  it "is not valid if the phone number is not 10 chars" do
+    expect(subject.phone.length).to be(10)
+  end
+  it "is not valid if the phone number is not all digits" do
+    expect(subject.phone).to match(/^[0-9]*$/)
+  end
+  it "is not valid if the email address doesn't have a @" do
+    expect(subject.email).to match(/@/)
+  end
```

Comment:
The goal of these is to check whether certain bad inputs cause the model to become invalid. Instead, it seems you've copied the validations that belong in the model, not in the spec. Here's an example of how one test could look like.

```ruby
  it "is not valid if the phone number is not 10 chars" do
    subject.phone="123456789"
    expect(subject).to_not be_valid
  end
```

You may want to check with a mentor to get a better understanding on how to write tests.

Reply:
ok got it i will correct it. I was thinking emphasis is on adding validation